awx/main/models/credential.py

```diff
@@ -326,9 +326,14 @@ def needs_vault_password(self):
     @property
     def passwords_needed(self):
         needed = []
-        for field in ('ssh_password', 'become_password', 'ssh_key_unlock', 'vault_password'):
+        for field in ('ssh_password', 'become_password', 'ssh_key_unlock'):
             if getattr(self, 'needs_%s' % field):
                 needed.append(field)
+        if self.needs_vault_password:
+            if self.inputs.get('vault_id'):
+                needed.append('vault_password.{}'.format(self.inputs.get('vault_id')))
+            else:
+                needed.append('vault_password')
```

Comment:
@ryanpetrello this is something you specifically referred to doing (I wanted to do this at the same time as connecting needed creds to the cred list), and I think it takes care of it, because this is the same string format that tasks.py is expecting. Haven't tested _fully_, but I don't have a reason to say it won't work anymore.

Reply:
👏 I'll test this all post-merge, but this _should be_ all we need for multi vault to be called done.